src/Utils/resolveIdRefs.ts

```diff
@@ -0,0 +1,72 @@
+import { JSONSchema7Definition } from "json-schema";
+import { StringMap } from "./StringMap";
+import { isLocalRef } from "./isLocalRef";
+import { DEFINITION_OFFSET } from "./constants";
+
+/**
+ * Replace all `#/definitions/{definition-id}` and `$ref` in the schema with actual definition names.
+ */
+export function resolveIdRefs(
+    schema: JSONSchema7Definition,
+    idNameMap: Map<string, string>,
+    encodeRefs: boolean
+): JSONSchema7Definition {
+    if (!schema || typeof schema === "boolean") {
+        return schema;
+    }
+    const { $ref, allOf, oneOf, anyOf, not, properties, items, definitions, additionalProperties, ...rest } = schema;
+    const result: JSONSchema7Definition = { ...rest };
+    if ($ref) {
+        if (!isLocalRef($ref)) {
+            result.$ref = $ref;
+        } else {
+            // THE Money Shot.
```

Comment:
The only actual work performed by this function is here, in the next 3 lines.

Everything else is just recursively traversing the schema.